internal/utils/name_resolvers/multi_address_resolver.go

```diff
@@ -0,0 +1,56 @@
+package name_resolvers
+
+// To enable grpc clients to connect to multiple servers ("," delimited)
+// Refer documentation from https://github.com/grpc/grpc/blob/master/doc/naming.md
+// Refer examples from https://github.com/grpc/grpc-go/tree/master/examples/features/name_resolving
+
+import (
+	"google.golang.org/grpc/resolver"
+	"strings"
+)
+
+// The target endpoint needs to start with this scheme to get parsed as this resolver
+// Eg: multi:///127.0.0.1:8001,127.0.0.1:8002,127.0.0.1:8003
+const scheme = "multi"
+
+type multiAddress struct {
+}
+
+func (m multiAddress) Build(target resolver.Target, cc resolver.ClientConn, opts resolver.BuildOptions) (resolver.Resolver, error) {
+	addresses := parseTarget(target.Endpoint)
+	r := &multiAddressResolver{
+		target:    target,
+		cc:        cc,
+		addresses: addresses,
+	}
+	r.start()
+	return r, nil
+}
```

Comment:
A library already exists which offers this https://pkg.go.dev/github.com/Jille/grpc-multi-resolver why not use that? This is used by [k8s](sigs.k8s.io/kpng/client) folks

Reply:
thanks. replaced.